### PR TITLE
Dala 4125 introduce vsme framework

### PR DESCRIPTION
**Pull request by stewal19**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
  - [x] If this PR includes work on the frontend, at least one `@ts-nocheck` is removed. Additionally, there should not be any `@ts-nocheck` in files modified by this PR. If no `@ts-nocheck` are left: Celebrate :tada: :confetti_ball: type-safety and remove this entry. 
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [ ] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [ ] Documentation is updated as required
- [ ] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server. Testing of the new feature should (also) be done by a second, independent reviewer from the dev team

### CODE REVIEW

File: dataland-frontend/src/components/forms/CreateSmeDataset.vue [hunk 1]
```diff
@@ -171,21 +211,15 @@ export default defineComponent({
...
-        }
-      },
-      set() {
-        // IGNORED
-      },
+    reportingPeriodYear(): number {
+      if (this.reportingPeriod) {
+        return this.reportingPeriod.getFullYear();
+      }
+      return 0;
```
Comment (jhaerDfine) [inline dataland-frontend/src/components/forms/CreateSmeDataset.vue:222]: is there a way to use null/undefined as default?

File: dataland-frontend/src/components/forms/CreateSmeDataset.vue [hunk 2]
```diff
@@ -121,11 +138,19 @@ import SingleSelectFormField from "@/components/forms/parts/fields/SingleSelectF
...
-import { getFilledKpis } from "@/utils/DataPoint";
 import { getBasePrivateFrameworkDefinition } from "@/frameworks/BasePrivateFrameworkRegistry";
 import { type PrivateFrameworkDataApi } from "@/utils/api/UnifiedFrameworkDataApi";
+import PollutionEmissionFormField from "@/components/forms/parts/fields/PollutionEmissionFormField.vue";
+import SubsidiaryFormField from "@/components/forms/parts/fields/SubsidiaryFormField.vue";
+import YesNoBaseDataPointFormField from "@/components/forms/parts/fields/YesNoBaseDataPointFormField.vue";
+import FreeTextFormField from "@/components/forms/parts/fields/FreeTextFormField.vue";
+import RadioButtonsFormField from "@/components/forms/parts/fields/RadioButtonsFormField.vue";
+import WasteClassificationFormField from "@/components/forms/parts/fields/WasteClassificationFormField.vue";
+import SiteAndAreaFormField from "@/components/forms/parts/fields/SiteAndAreaFormField.vue";
```
Comment (jhaerDfine) [inline dataland-frontend/src/components/forms/CreateSmeDataset.vue:151]: some issues with this formField.
filling out (only) the required fields doesnt let me upload the report, even though im allowed to click the button.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/PrivateDataManager.kt [hunk 3]
```diff
@@ -269,18 +269,18 @@ class PrivateDataManager(
     }
 
     /**
-     * Retrieves a private sme data object from the private storage
+     * Retrieves a private vsme data object from the private storage
      * @param dataId the dataId of the dataset to be retrieved
      * @param correlationId the correlationId of the request
-     * @return the sme dataset
+     * @return the v dataset
```
Comment (jhaerDfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/PrivateDataManager.kt:275]: v -> vsme

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt [hunk 4]
```diff
@@ -77,66 +80,81 @@ class Sme {
     }
 
     @Test
-    fun `post SME data and check its meta info persistence and that data is not even accessible to Dataland admins `() {
-        val smeData = setPowerConsumptionFileReference(testSmeData, hashAlpha)
-        val companyAssociatedSmeData = CompanyAssociatedDataSmeData(companyId, "2022", smeData)
-        val dataMetaInfoInResponse = postSmeDataset(companyAssociatedSmeData, listOf(dummyFileAlpha))
+    fun `post VSME data check its meta info persistence and that data is not even accessible to Dataland admins `() {
+        val vsmeData = setReferencedReports(testVsmeData, FileInfos(hashAlpha, fileNameAlpha))
+        val companyAssociatedvsmeData = CompanyAssociatedDataVsmeData(companyId, "2022", vsmeData)
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt:85]: companyAssociatedVsmeData (camle case)

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt [hunk 5]
```diff
@@ -77,66 +80,81 @@ class Sme {
...
     }
 
     @Test
-    fun `post SME data with documents and check if data and documents can be retrieved by the data owner`() {
-        val smeData = setPowerConsumptionFileReference(testSmeData, hashAlpha)
-        val companyAssociatedDataSmeData = CompanyAssociatedDataSmeData(companyId, "2023", smeData)
-        val dataMetaInfoInResponse = postSmeDataset(companyAssociatedDataSmeData, listOf(dummyFileAlpha, dummyFileBeta))
+    fun `post VSME data with documents and check if data and documents can be retrieved by the data owner`() {
+        val vsmeData = setReferencedReports(testVsmeData, FileInfos(hashAlpha, fileNameAlpha))
+        val companyAssociatedDataSmeData = CompanyAssociatedDataVsmeData(companyId, "2023", vsmeData)
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt:107]: companyAssociatedDataSmeData -> companyAssociatedData**V**smeData

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt [hunk 6]
```diff
@@ -77,66 +80,81 @@ class Sme {
...
     }
 
     @Test
-    fun `post two SME datasets for the same reporting period and company and assert correct handling`() {
-        var smeData = setPowerConsumptionFileReference(testSmeData, null)
-        val companyAssociatedSmeDataAlpha = generateSmeDataWithSetNumberOfEmployees(companyId, "2022", smeData, 1)
-        val dataIdAlpha = postSmeDataset(companyAssociatedSmeDataAlpha).dataId
+    fun `post two VSME datasets for the same reporting period and company and assert correct handling`() {
+        var vsmeData = setReferencedReports(testVsmeData, null)
+        val companyAssociatedSmeDataAlpha =
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt:127]: sme or vsme?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt [hunk 7]
```diff
@@ -77,66 +80,81 @@ class Sme {
...
+            BigDecimal(1),
+            retrievedCompanyAssociatedSmeDataAlpha?.data?.basic?.workforceGeneralCharacteristics
+                ?.numberOfEmployeesInHeadcount,
         )
-        assertEquals(1, retrievedCompanyAssociatedSmeDataAlpha?.data?.general?.basicInformation?.numberOfEmployees)
 
-        smeData = setPowerConsumptionFileReference(testSmeData, hashAlpha)
-        val companyAssociatedSmeDataBeta = generateSmeDataWithSetNumberOfEmployees(companyId, "2022", smeData, 2)
+        vsmeData = setReferencedReports(testVsmeData, FileInfos(hashAlpha, fileNameAlpha))
+        val companyAssociatedSmeDataBeta =
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt:142]: vsme or sme?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt [hunk 8]
```diff
@@ -150,57 +168,43 @@ class Sme {
     }
 
     @Test
-    fun `post an SME dataset with duplicate file and assert that the downloaded file is unique and correct`() {
-        val smeData = setPowerConsumptionFileReference(testSmeData, hashAlpha)
-        val companyAssociatedSmeData = CompanyAssociatedDataSmeData(companyId, "2022", smeData)
-        val dataId = postSmeDataset(companyAssociatedSmeData, listOf(dummyFileAlpha, dummyFileAlpha)).dataId
+    fun `post an VSME dataset with duplicate file and assert that the downloaded file is unique and correct`() {
+        val vsmeData = setReferencedReports(testVsmeData, FileInfos(hashAlpha, fileNameAlpha))
+        val companyAssociatedSmeData = CompanyAssociatedDataVsmeData(companyId, "2022", vsmeData)
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt:173]: vsme or sme?

File: dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt [hunk 9]
```diff
@@ -150,57 +168,43 @@ class Sme {
...
-                    numberOfEmployees = numberOfEmployees,
-                ),
-            ),
-        )
-    }
-
-    private fun postSmeDataset(
-        companyAssociatedDataSmeData: CompanyAssociatedDataSmeData,
+    private fun postVsmeDataset(
+        companyAssociatedDataSmeData: CompanyAssociatedDataVsmeData,
```
Comment (jhaerDfine) [inline dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Vsme.kt:207]: Sme or vsme?

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/SmeSimpleCustomComponentBase.kt [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.typescript.TypeScriptImport
+
+/**
+ * Represents the SmeSimpleCustomComponentBase component
+ * @param viewFormattingFunctionName the name of the formatting function for the view page in SmeDisplayValueGetters.ts
+ */
+open class SmeSimpleCustomComponentBase(
```
Comment (jhaerDfine) [inline dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/SmeSimpleCustomComponentBase.kt:21]: do you want to keept the class names as sme... and not vsme?

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/VsmeSimpleCustomComponentBase.kt [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+import org.dataland.frameworktoolbox.intermediate.datapoints.NoDocumentSupport
+import org.dataland.frameworktoolbox.specific.fixturegenerator.elements.FixtureSectionBuilder
+import org.dataland.frameworktoolbox.specific.uploadconfig.elements.UploadCategoryBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.SectionConfigBuilder
+import org.dataland.frameworktoolbox.specific.viewconfig.elements.getTypescriptFieldAccessor
+import org.dataland.frameworktoolbox.specific.viewconfig.functional.FrameworkDisplayValueLambda
+import org.dataland.frameworktoolbox.utils.typescript.TypeScriptImport
+
+/**
+ * Represents the SmeSimpleCustomComponentBase component
```
Comment (jhaerDfine) [inline dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/VsmeSimpleCustomComponentBase.kt:18]: VsmeSimple

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/VsmeSimpleCustomComponentBase.kt [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+ * @param viewFormattingFunctionName the name of the formatting function for the view page in SmeDisplayValueGetters.ts
```
Comment (jhaerDfine) [inline dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/VsmeSimpleCustomComponentBase.kt:19]: sme -> vsme

File: dataland-frontend/src/components/resources/dataTable/conversion/MultiLayerDataTableFieldConverter.ts [hunk 13]
```diff
@@ -43,6 +44,8 @@ const formFieldValueGetterFactoryMap: { [key: string]: ValueGetterFactory } = {
   IntegerExtendedDataPointFormField: numberDataPointValueGetterFactory,
   BigDecimalExtendedDataPointFormField: numberDataPointValueGetterFactory,
   CurrencyDataPointFormField: currencyDataPointValueGetterFactory,
+  SubsidiaryFormField: getModalGetterFactory("listOfSubsidiary", smeModalColumnHeaders),
+  PollutionEmissionFormField: getModalGetterFactory("pollutionEmission", smeModalColumnHeaders),
```
Comment (jhaerDfine) [inline dataland-frontend/src/components/resources/dataTable/conversion/MultiLayerDataTableFieldConverter.ts:48]: sme or vsme?

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/SmeSimpleCustomComponentBase.kt (hunk 10 above)
Comment (stewal19) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/vsme/custom/SmeSimpleCustomComponentBase.kt]: Sollen am Ende alle vsme heißen. Soweit ich gesehen hab ist das hier auch korrgiert worden

File: dataland-frontend/src/components/forms/CreateSmeDataset.vue (hunk 2 above)
Comment (stewal19) [reply dataland-frontend/src/components/forms/CreateSmeDataset.vue:151]: Can you check again, we have aligned the required fields in the frontend with the required fields in the backend.

File: dataland-frontend/src/components/forms/CreateSmeDataset.vue (hunk 1 above)
Comment (stewal19) [reply dataland-frontend/src/components/forms/CreateSmeDataset.vue:212]: Looking into this

File: dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/sme/custom/SmeSiteAndAreaComponent.kt [hunk 14]
```diff
@@ -0,0 +1,40 @@
+package org.dataland.frameworktoolbox.frameworks.sme.custom
+
+import org.dataland.frameworktoolbox.intermediate.FieldNodeParent
+import org.dataland.frameworktoolbox.intermediate.components.requireDocumentSupportIn
+import org.dataland.frameworktoolbox.intermediate.datapoints.NoDocumentSupport
+import org.dataland.frameworktoolbox.specific.datamodel.TypeReference
+import org.dataland.frameworktoolbox.specific.datamodel.elements.DataClassBuilder
+
+/**
+ * Represents the sitesandareas component for vsme
```
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/sme/custom/SmeSiteAndAreaComponent.kt]: Changed

File: dataland-frontend/tests/e2e/specs/sme/DataIntegrity.ts [hunk 15]
```diff
@@ -1,116 +1,187 @@
...
-    smeFixtureForTest = getPreparedFixture("Sme-dataset-with-no-null-fields", preparedFixturesSme);
-  });
-});
-
+/*TODO
+- Eröffne backlog item um EDIT Funktionalität wieder einzuführen
+ */
 describeIf(
-  "As a user, I expect to be able to edit and submit Sme data via the upload form",
+  "As a user, I expect to be able to edit and submit Vsme data via the upload form",
```
Comment (jhaerDfine) [inline dataland-frontend/tests/e2e/specs/sme/DataIntegrity.ts:19]: change the folder name or move the file to the (new) vsme folder of this file?

File: dataland-frontend/src/components/resources/frameworkDataSearch/frameworkPanel/MultiLayerDataTableFrameworkPanel.vue [hunk 16]
```diff
@@ -32,6 +32,9 @@
       :ariaLabel="`Datasets of the ${frameworkDisplayName} framework`"
     />
   </div>
+  <div v-if="status == 'InsufficientRights'">
+    <h1>Sorry! You have insufficient rights to view this ressource.</h1>
```
Comment (jhaerDfine) [inline dataland-frontend/src/components/resources/frameworkDataSearch/frameworkPanel/MultiLayerDataTableFrameworkPanel.vue:36]: typo: ressource -> resource

Comment (stewal19) [reply dataland-frontend/src/components/resources/frameworkDataSearch/frameworkPanel/MultiLayerDataTableFrameworkPanel.vue]: changed

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/PrivateDataManager.kt (hunk 3 above)
Comment (stewal19) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/PrivateDataManager.kt:275]: Changed

File: dataland-framework-toolbox/inputs/vsme/vsme.xlsx
[the remark this answers is not in the document]
Comment (stewal19) [reply dataland-framework-toolbox/inputs/vsme/vsme.xlsx:1]: Should be fixed now

**Pull request merged**